### PR TITLE
Fix csm translate bug v3.7.2

### PR DESCRIPTION
**Pull request by troublemaker52025**
Re: #

### Changelog

*

-------

### Continuous Integration

This pull request:

* [x] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

**Comment by troublemaker52025** [comment]
@cocos-robot run test cases

File: editor/assets/chunks/builtin/functionalities/shadow-map.chunk [hunk 1]
```diff
@@ -250,89 +250,122 @@ float CCGetLinearDepth(vec3 worldPos) {
...
+        return true;
+      }
+      if (clipPos.y >= maxRange) {
+        ratio = 1.0 - (clipPos.y - maxRange) * thresholdInvert;
+        return true;
+      }
+      return false;
+    }
+
+    bool CCHasLevel(int layer, vec3 worldPos) {
```
Comment (jk20012001) [inline editor/assets/chunks/builtin/functionalities/shadow-map.chunk]: CCHasCSMLayer

File: editor/assets/chunks/builtin/functionalities/shadow-map.chunk [hunk 2]
```diff
@@ -250,89 +250,122 @@ float CCGetLinearDepth(vec3 worldPos) {
...
-          clipPos.y <= minRange || clipPos.y >= maxRange) {
-        if (clipPos.x >= layerThreshold && clipPos.x <= cc_csmSplitsInfo.x) {
-          ratio = (clipPos.x - layerThreshold) / (cc_csmSplitsInfo.x - layerThreshold);
-        }
-        if (clipPos.x >= maxRange && clipPos.x <= 1.0 - layerThreshold) {
-          ratio = (clipPos.x - maxRange) / (cc_csmSplitsInfo.x - layerThreshold);
-        }
-        if (clipPos.y >= 0.0 && clipPos.y <= cc_csmSplitsInfo.x) {
-          ratio = min((clipPos.y - layerThreshold) / (cc_csmSplitsInfo.x - layerThreshold), 1.0);
+      bool hasLevel = false;
```
Comment (jk20012001) [inline editor/assets/chunks/builtin/functionalities/shadow-map.chunk:279]: hasLayer

File: editor/assets/chunks/builtin/functionalities/shadow-map.chunk [hunk 3]
```diff
@@ -250,89 +250,122 @@ float CCGetLinearDepth(vec3 worldPos) {
...
+          shadowViewDir1 = cc_csmViewDir1[i].xyz;
+          shadowViewDir2 = cc_csmViewDir2[i].xyz;
         }
-        return true;
       }
-      return false;
     }
 
-    int CCGetCSMLevel(out bool hasNextTransition, out bool transitionArea, out float transitionRatio, out vec4 csmPos, out vec4 shadowProjDepthInfo, out vec4 shadowProjInfo, out vec3 shadowViewDir0, out vec3 shadowViewDir1, out vec3 shadowViewDir2, vec3 worldPos, int nextLayer)
+    int CCGetCSMLevel(out bool isTransitionArea, out highp float transitionRatio, out vec4 csmPos, out vec4 shadowProjDepthInfo, out vec4 shadowProjInfo, out vec3 shadowViewDir0, out vec3 shadowViewDir1, out vec3 shadowViewDir2, vec3 worldPos)
```
Comment (jk20012001) [inline editor/assets/chunks/builtin/functionalities/shadow-map.chunk:313]: Level全局替换成Layer

**Review by jk20012001 — APPROVED** [review]

**Pull request merged**